.circleci/config.yml

```diff
@@ -1,15 +1,79 @@
 version: 2.1
 jobs:
-  build:
+  build-dev:
     docker:
-      - image: circleci/golang:1.12
+      - image: liftbridge/liftbridge-circleci:0.1.0
         environment:
           GO111MODULE: "on"
-
+    working_directory: "~/project"
+    steps:
+      - checkout
+      - run:
+          name: Build dev binary
+          command: |
+            make build-dev
+      - persist_to_workspace:
+          root: .
+          paths:
+            - liftbridge-dev
+            - docker/
+  test:
+    docker:
+      - image: liftbridge/liftbridge-circleci:0.1.0
+        environment:
+          GO111MODULE: "on"
+    working_directory: "~/project"
     steps:
       - checkout
-      - run: sudo apt-get update -yqq && sudo apt-get install -yqq bzr
-      - run: go get github.com/mattn/goveralls
-      - run: go test -v .
-      - run: go test -v -coverprofile=coverage.txt ./server/...
-      - run: goveralls -coverprofile=coverage.txt -service=circle-ci -repotoken=$COVERALLS_REPO_TOKEN
+      - run:
+          name: Install dependencies
+          command: |
+            go get github.com/mattn/goveralls
+      - run:
+          name: Test
+          command: |
+            go test -v .
+            go test -v -coverprofile=coverage.txt ./server/...
+            goveralls -coverprofile=coverage.txt -service=circle-ci -repotoken=$COVERALLS_REPO_TOKEN
+
+  build-and-push-standalone-dev-image:
+    docker:
+      - image: circleci/buildpack-deps:stretch
+    working_directory: "~/project"
+    steps:
+      - setup_remote_docker
+      - attach_workspace:
+          at: ~/project
+      - run:
+          name: set image version
+          command: |
+            if [ -n "$CIRCLE_TAG" ]; then
+                echo "export VERSION='${CIRCLE_TAG}'" >> $BASH_ENV
+            else
+                echo "export VERSION='${CIRCLE_SHA1}'" >> $BASH_ENV
+            fi
+      - run:
+          name: build image
+          command: |
+            docker build -t "liftbridge/standalone-dev:${VERSION}" -f docker/dev-standalone/Dockerfile .
+            docker tag "liftbridge/standalone-dev:${VERSION}" 'liftbridge/standalone-dev:latest'
```

Comment:
I wonder if we should push a "production" image if there is a tag? Or maybe it would be better to just keep that a separate release process.

Reply:
I think it's a good idea and it's pretty easy to set up. I have something like that set up on the Clojure client: if the tag is a semver, it publishes a release. This is the filter config: https://github.com/caioaao/clj-liftbridge/blob/94971bfc8c5962fb1d24cf41f0f395c4fc267204/.circleci/config.yml#L91-L95

Reply:
also, might be a good idea to add the `test` step as a requirement to merge PRs

Reply:
We can address this in a future PR.